### ISSUE TITLE
Attention Decoder

### ISSUE THREAD

**Issue by mhnatiuk** [tensorflow/tensorflow#8775]
Hi,
@ebrevdo Eugene speaks here https://www.youtube.com/watch?v=RIR_-Xlbp7s (29:50) about a class AttentionDecoder and says it's under development (as of Tensorflow Summit date), however I can't find this class anywhere in the master/r1.1/r1.0 tensorflow code. Do you know when it will be added?

**Comment by ebrevdo** [comment]
We need to update that video somehow.  It turns out it's cleaner to
implement attention as an RNNCell wrapper rather than decoder.  This way it
can be combined with other decoders easily.  See tensorflow master branch,
tf.contrib.seq2seq.AttentionWrapper.  It's pretty feature complete, and can
be used with other RNNCells, wrappers, and the BasicDecoder.

**Comment by mhnatiuk** [comment]
Ok I tired with LSTMBasicCell and got "state has no attribute attention" exception.. looks like a bug

**Comment by mhnatiuk** [comment]
I will review and report back

**Comment by RobRomijnders** [comment]
I'm running into that same issue. How did you solve it?

**Comment by mhnatiuk** [comment]
I switched back to using TrainingHelper instead of ScheduledEmbeddingHelper, and feeded embedded inputs by looking up embeddings using tf.nn.embedding_lookup. It works, now I run into other problem, the shape of decoder_outputs.rnn_output is (AFAIK) *dynamic* and it's not possible to feed it into tf.contrib.seq2seq.sequence_loss, since it expects TARGETS and LOGITS to have compatible shapes
